src/features/discord/messages/functions/commands/flags/discord-command-get-flag-name.ts

```diff
@@ -0,0 +1,35 @@
+import _ from "lodash";
+import { discordCommandRemoveFlagPrefix } from "./discord-command-remove-flag-prefix";
+
+/**
+ * @description
+ * Extract the flag name from a message flag
+ *
+ * @example
+ * discordCommandGetFlagName('')          => null
+ * discordCommandGetFlagName('-f')        => f
+ * discordCommandGetFlagName('-f=')       => f
+ * discordCommandGetFlagName('-f=dummy')  => f
+ * discordCommandGetFlagName('--f')       => f
+ * discordCommandGetFlagName('--f=')      => f
+ * discordCommandGetFlagName('--f=dummy') => f
+ *
+ * @param {Readonly<string>} messageFlag A flag as a message
+ * @param {Readonly<boolean>} toLowerCase Return the flag name to lower case
+ *
+ * @return {string | null} a string when the flag name exists
+ */
+export function discordCommandGetFlagName(
+  messageFlag: Readonly<string>,
```

Comment:
@Vincent-Valembois adding a new type "MessageFlag" would be easier to understand the difference here.
I am not a big fan of this thing here since it works with string to I do not have a lot of control over it but having functions handling a pattern like "--x=y" made this PR much simplier